test/winter_onboarding_2021/fleet_management/handlers/cab_test.clj

```diff
@@ -28,4 +31,27 @@
                       :data invalid-cab}
               :headers {"Location" "/cabs/new"}
               :body ""}
-             (cab/create {:multipart-params invalid-cab}))))))
+             (cab-handler/create {:multipart-params invalid-cab}))))))
+
+(deftest list-cabs-handler
+  (testing "Should return a list of 3 rows of cabs"
+    (let [cabs (factories/create-cabs 3)]
+      (doall (map cab/create cabs))
+      (is (= 3 (count (hf/hiccup-find [:tbody :tr] (cab-handler/get-cabs {})))))
+      (is (= 0 (count (hf/hiccup-find [:a] (cab-handler/get-cabs {})))))))
+
+  (testing "Should return a list of 12 rows of cabs with next Page link"
+    (let [cabs (factories/create-cabs 12)]
+      (doall (map cab/create cabs))
+      (is (= 10 (count (hf/hiccup-find [:tbody :tr] (cab-handler/get-cabs {})))))
+      (is (= 1 (count (hf/hiccup-find [:a] (cab-handler/get-cabs {})))))))
+
+  (testing "Should return 8 rows of cabs in page number 2"
+    (is (= 5 (count (hf/hiccup-find [:tbody :tr] (cab-handler/get-cabs {:query-params
```

Comment:
You can use ring-mock to mock the requests.

Reply:
Ring-mock will need to go through the wrap-params middleware. May increase complexity